packages/edit-site/src/components/dataviews/sidebar-content.js

```diff
@@ -1,4 +1,205 @@
+/**
+ * WordPress dependencies
+ */
+import {
+	__experimentalItemGroup as ItemGroup,
+	DropdownMenu,
+	MenuGroup,
+	MenuItem,
+	Modal,
+	Button,
+	TextControl,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+import { moreVertical, plus, page, columns } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
+import { useContext, useState, useEffect, useMemo } from '@wordpress/element';
+import { useDispatch, useSelect } from '@wordpress/data';
+import { store as coreStore } from '@wordpress/core-data';
+
+/**
+ * Internal dependencies
+ */
+import SidebarNavigationItem from '../sidebar-navigation-item';
+import DataviewsContext from '../dataviews/context';
+import { DEFAULT_VIEWS } from './provider';
+
+function DataViewItem( { dataviewId, setCurrentViewId, isActive } ) {
+	const { dataview } = useSelect(
+		( select ) => {
+			const { getEditedEntityRecord } = select( coreStore );
+			return {
+				dataview: getEditedEntityRecord(
+					'postType',
+					'wp_dataviews',
+					dataviewId
+				),
+			};
+		},
+		[ dataviewId ]
+	);
+	const icon = useMemo( () => {
+		const viewContent = JSON.parse( dataview.content );
+		switch ( viewContent.type ) {
+			case 'list':
+				return page;
+			case 'grid':
+				return columns;
+		}
+	}, [ dataview.content ] );
+	return (
+		<SidebarNavigationItem
+			icon={ icon }
+			onClick={ () => {
+				setCurrentViewId( dataviewId );
+			} }
+			aria-current={ isActive ? 'true' : undefined }
+			suffix={
+				<DropdownMenu
+					icon={ moreVertical }
+					label={ __( 'Actions' ) }
+					toggleProps={ {
+						style: {
+							color: 'inherit',
+						},
+					} }
+				>
+					{ ( { onClose } ) => (
+						<MenuGroup>
+							<MenuItem onClick={ () => onClose() }>
+								{ __( 'Rename' ) }
+							</MenuItem>
+							<MenuItem onClick={ () => onClose() }>
+								{ __( 'Delete' ) }
+							</MenuItem>
+						</MenuGroup>
+					) }
+				</DropdownMenu>
+			}
+		>
+			{ dataview.title }
+		</SidebarNavigationItem>
+	);
+}
+
+function AddNewItem( { type, taxonomyId } ) {
+	const { saveEntityRecord } = useDispatch( coreStore );
+
+	const [ isAdding, setIsAdding ] = useState( false );
```

Comment:
Why do we need this boolean in addition to `dataViewIsSaving`?

Reply:
Oh I guess, this one controls the visibility of the modal.

Reply:
Exactly control modal visibility.